src/app/shared/services/filter-map/filter-map.service.ts

```diff
@@ -0,0 +1,139 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { Marker, SongFilter } from '../../interfaces/map-marker';
+import { Store } from '@ngxs/store';
+import { MapState } from '../../../store/map/map.state';
+import { API_URL, StatEndpoints } from '../../config/endpoints/stat-endpoints';
+import { catchError } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class FilterMapService {
+  constructor(
+    private http: HttpClient,
+    private store: Store
+  ) {}
+
+  countSongsByOption(option: string): number {
+    const filteredMarkers = this.store.selectSnapshot(MapState.getFilteredMarkerList);
+
+    const foundMarkers = filteredMarkers.filter((marker) => {
+      return (
+        marker.location.country.toLowerCase().includes(option.toLowerCase()) ||
+        marker.location.region.toLowerCase().includes(option.toLowerCase()) ||
+        marker.genre_cycle.toLowerCase().includes(option.toLowerCase()) ||
+        marker.found.toLowerCase().includes(option.toLowerCase()) ||
+        marker.location.district_center.toLowerCase().includes(option.toLowerCase())
+      );
+    });
+
+    return foundMarkers.length;
+  }
+
+  filteredMarker(selectOptions: SongFilter): Marker[] {
```

Comment:
Naming Convention for Functions
JavaScript function names should be in camel case and use **descriptive nouns and verbs as prefixes**. This makes it clear what the function does and what kind of value it returns.